airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -82,7 +108,194 @@ public JsonNode toDatabaseConfig(JsonNode mssqlConfig) {
+            LOGGER.warn(String.format("Skipping check for whether the SQL Server Agent is running, SQLServerException thrown: '%s'",
+                e.getMessage()));
+          } else {
+            throw e;
+          }
+        }
+      });
+
+      // check that snapshot isolation is allowed
+      checkOperations.add(database -> {
+        List<JsonNode> queryResponse = database.query(connection -> {
+          final String sql = "SELECT name, snapshot_isolation_state FROM sys.databases WHERE name = ?";
+          PreparedStatement ps = connection.prepareStatement(sql);
+          ps.setString(1, config.get("database").asText());
+          LOGGER.info(String.format("Checking that snapshot isolation is enabled on database '%s' using the query: '%s'",
+              config.get("database").asText(), sql));
+          return ps;
+        }, JdbcUtils::rowToJson).collect(toList());
+
+        if (queryResponse.size() < 1) {
+          throw new RuntimeException(String.format(
+              "Couldn't find '%s' in sys.databases table. Please check the spelling and that the user has relevant permissions (see docs).",
+              config.get("database").asText()));
+        }
+
+        if (queryResponse.get(0).get("snapshot_isolation_state").asInt() != 1) {
+          throw new RuntimeException(String.format(
+              "Detected that snapshot isolation is not enabled for database '%s'. MSSQL CDC relies on snapshot isolation. "
+                  + "Please check the documentation on how to enable snapshot isolation on MS SQL Server.",
+              config.get("database").asText()));
+        }
+      });
+
+    }
+
+    return checkOperations;
+  }
+
+  @Override
+  public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(JdbcDatabase database,
+                                                                             ConfiguredAirbyteCatalog catalog,
+                                                                             Map<String, TableInfo<CommonField<JDBCType>>> tableNameToTable,
+                                                                             StateManager stateManager,
+                                                                             Instant emittedAt) {
+    JsonNode sourceConfig = database.getSourceConfig();
+    if (isCdc(sourceConfig) && shouldUseCDC(catalog)) {
+      LOGGER.info("using CDC: {}", true);
+      AirbyteDebeziumHandler handler = new AirbyteDebeziumHandler(sourceConfig, MssqlCdcTargetPosition.getTargetPostion(database),
+          MssqlCdcProperties.getDebeziumProperties(), catalog, true);
+      return handler.getIncrementalIterators(new MssqlCdcSavedInfoFetcher(stateManager.getCdcStateManager().getCdcState()),
+          new MssqlCdcStateHandler(stateManager), new MssqlCdcConnectorMetadataInjector(), emittedAt);
+    } else {
+      LOGGER.info("using CDC: {}", false);
```

Comment:
just making it easier to read
```suggestion
      LOGGER.info("not using CDC");
```